src/shared/lib/resources/user.ts

```diff
@@ -37,12 +38,12 @@ export interface User {
   name: string;
   email: string;
   jobTitle: string;
-  avatarImageFile?: FileRecord;
-  notificationsOn?: Date;
-  acceptedTerms?: Date;
+  avatarImageFile: FileRecord | null;
+  notificationsOn: Date | null;
+  acceptedTerms: Date | null;
   idpUsername: string;
-  deactivatedOn?: Date;
-  deactivatedBy?: Id;
+  deactivatedOn: Date | null;
+  deactivatedBy: Id | null;
```

Comment:
@sutherlanda TODO: investigate whether this change needs to be made across the board.